tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_section_header_body_mismatch.py

```diff
@@ -0,0 +1,146 @@
+                    skip_body_listing=False,
+                    expected_code="ef000101000802000100030400040000800001000000003050003050000bad60A7",  # noqa: E501
+                    expected_exception=[
+                        EOFException.INVALID_TYPE_SECTION_SIZE,
+                        EOFException.INVALID_SECTION_BODIES_SIZE,
+                    ],
+                ),
+                id="drop_code_header",
+            ),
+            pytest.param(
+                dict(
+                    skip_header_listing=False,
+                    skip_body_listing=True,
+                    expected_code="ef0001010008020002000300030400040000800001000000003050000bad60A7",  # noqa: E501
+                    expected_exception=[
+                        EOFException.UNREACHABLE_CODE_SECTIONS,
+                        EOFException.TOPLEVEL_CONTAINER_TRUNCATED,
+                    ],
+                ),
+                id="drop_code_section",
+            ),
+            pytest.param(
+                dict(
+                    skip_header_listing=False,
+                    skip_body_listing=False,
+                    expected_code="ef0001010008020002000300030400040000800001000000003050003050000bad60A7",  # noqa: E501
+                    expected_exception=EOFException.UNREACHABLE_CODE_SECTIONS,
+                ),
+                id="layout_ok_code_bad",
+            ),
+            pytest.param(
+                dict(
+                    skip_header_listing=True,
+                    skip_body_listing=True,
+                    skip_types_body_listing=True,
+                    expected_code="ef0001010008020001000304000400008000013050000bad60a7",
+                    expected_exception=[
+                        EOFException.INVALID_TYPE_SECTION_SIZE,
+                        EOFException.INVALID_SECTION_BODIES_SIZE,
+                    ],
+                ),
+                id="drop_types_header",
+            ),
+            pytest.param(
+                dict(
+                    skip_header_listing=True,
+                    skip_body_listing=True,
+                    skip_types_body_listing=True,
+                    skip_types_header_listing=True,
+                    expected_code="ef0001010004020001000304000400008000013050000bad60a7",
+                    expected_exception=None,
+                ),
+                id="drop_everything",
+            ),
+        ],
+    )
+)
+def test_code_section_header_body_mismatch(
+    eof_test: EOFTestFiller,
+    skip_header_listing: bool,
+    skip_body_listing: bool,
+    skip_types_body_listing: bool,
+    skip_types_header_listing: bool,
+    expected_code: str,
+    expected_exception: EOFExceptionInstanceOrList | None,
+):
+    """
+    Inconsistent number of code sections (between types and code)
+    """
+    eof_code = Container(
+        name="EOF1I0018",
+        sections=[
+            Section.Code(
+                code=Op.ADDRESS + Op.POP + Op.STOP,
```

Comment:
Could we try `Op.CALLF[1] + Op.STOP` instead? This might get rid of `EOFException.UNREACHABLE_CODE_SECTIONS` and replace it with the correct exception.

Reply:
this actually intoduce another exception of unreachable code section because there is test case when the other section bytes are skipped.